Edit Contact Event Command

nicrandomlee opened the pull request:
### Edit Contact Calendar Event

Edits the details of an event in a contact's calendar

Format: `editContactEvent PERSON_INDEX EVENT_INDEX [d/DESCRIPTION] [ts/NEW_START_DATE_TIME][te/NEW_END_DATE_TIME]`

Edits `EVENT_INDEX` event of the `PERSON_INDEX` person in the address book with the given fields.

Example: `editContactEvent 1 1 d/Nap`, `editContactEvent 2 3 ts/2023-10-10 10:00 te/2023-10-12 15:00`

* Note that all edited fields are optional, but there must be at least 1 edited field.

nicrandomlee committed:
Merge branch 'master' of https://github.com/nicrandomlee/tp

nicrandomlee committed:
Merge branch 'AY2324S1-CS2103-F13-4:master' into master

nicrandomlee committed:
Merge branch 'AY2324S1-CS2103-F13-4:master' into master

nicrandomlee committed:
EditContactEvent Checkpoint

Added base non-functional code for EditContactEventCommand
and EditContactEventCommandParser

nicrandomlee committed:
Add Checkpoint

Add checkpoint for EditContactEventCommand and
EditContactEventCommandParser

nicrandomlee committed:
Save checkpoint

Added dummy EditEventDescriptor.
Added getter for EventPeriod.
Added getter for event list.

nicrandomlee committed:
Add checkpoint

Updated EditContactEventCommandParser
Added new helper method for ParserUtil

nicrandomlee committed:
Fix Checkstyle

nicrandomlee committed:
Add Checkpoint

Updated EditContactEventCommand and its parser
Changed visibility of EventPeriod's second constructor

nicrandomlee committed:
Fix bugs

Update typos for EditContactEventCommand
Changed visibility of EventPeriod constructor

nicrandomlee committed:
Add testcase

nicrandomlee committed:
Update test case

nicrandomlee committed:
Merge branch 'AY2324S1-CS2103-F13-4:master' into master

nicrandomlee committed:
Merge branch 'master' into branch-Misc-Calendar-Commands-NL

nicrandomlee committed:
Update Parser

nicrandomlee committed:
Merge branch 'AY2324S1-CS2103-F13-4:master' into master

nicrandomlee committed:
Add Test Case

Added test case for UniMateParser

nicrandomlee committed:
Add test case for EditContactEventCommandParser

nicrandomlee committed:
Update Checkstyle

nicrandomlee committed:
Change Line Formatting

nicrandomlee committed:
Add EditContactEventCommand test case

nicrandomlee committed:
Merge branch 'AY2324S1-CS2103-F13-4:master' into master

nicrandomlee committed:
Fix Checkstyle

Removed unused imports for EditPersonDescriptor
Added javadocs for EditPersonDescriptor

nicrandomlee committed:
Add EditContactEventCommand test case

nicrandomlee committed:
Fix checkstyle

Removed unused import

nicrandomlee committed:
Fix bug

Added guard clause for user exceeding index of event list

nicrandomlee committed:
Update User Guide

nicrandomlee committed:
Fix Checkstyle

nicrandomlee committed:
Revert Person class changes

nicrandomlee committed:
Merge branch 'master' into branch-Misc-Calendar-Commands-NL

# Conflicts:
#	src/main/java/seedu/address/logic/parser/ParserUtil.java
#	src/main/java/seedu/address/logic/parser/UniMateParser.java
#	src/test/java/seedu/address/logic/parser/UniMateParserTest.java

nicrandomlee committed:
Fix merge conflict

Fallman2 commented on src/main/java/seedu/address/logic/commands/EditContactEventCommand.java:
```diff
@@ -0,0 +1,270 @@
+     * @param eventIndex            The index of the event to be updated in the calendar's event list.
+     * @param editEventDescriptor   The descriptor containing the new information for the event.
+     * @return                      The updated event list.
+     */
```
Fix Javadoc alignment

Fallman2 reviewed (approved):
Generally LGTM

nicrandomlee committed:
Update javadoc

Updated javadoc to standardize format

nicrandomlee committed:
Remove unnecessary method

nicrandomlee committed:
Add EditContactEventCommand test case

nicrandomlee committed:
Add test case

nicrandomlee committed:
Add test case

nicrandomlee committed:
Fix Checkstyle violation

nicrandomlee committed:
Add test case

Increase code coverage to 90%

nicrandomlee committed:
Update User Guide

nicrandomlee committed:
Merge branch 'AY2324S1-CS2103-F13-4:master' into master

nicrandomlee committed:
Fix checkstyle

nicrandomlee committed:
Merge branch 'master' into branch-Misc-Calendar-Commands-NL

Fallman2 commented on docs/UserGuide.md:
```diff
@@ -219,9 +219,21 @@ Format: `viewEvents`
+
+Edits the details of an event in a contact's calendar
+
+Format: `editContactEvent PERSON_INDEX EVENT_INDEX [d/DESCRIPTION] [ts/NEW_START_DATE_TIME][te/NEW_END_DATE_TIME]`
```
Add spacing between each parameter

Fallman2 commented on src/main/java/seedu/address/logic/commands/EditContactEventCommand.java:
```diff
@@ -0,0 +1,270 @@
+    public EditContactEventCommand(ArrayList<Index> indexArrayList, EditEventDescriptor editEventDescriptor) {
+        requireNonNull(indexArrayList.get(0));
+        requireNonNull(indexArrayList.get(1));
+        requireNonNull(editEventDescriptor);
```
RequireAllNonNull

Fallman2 commented on src/main/java/seedu/address/logic/commands/EditContactEventCommand.java:
```diff
@@ -0,0 +1,270 @@
+     * Creates and returns a {@code Person} with the details of {@code personToEdit}, editing the event with index
+     * {@code eventIndex} with {@code editEventDescriptor}.
+     */
+    private static Person createPersonsEditedEvent(Index eventIndex,
+                                                   Person personToEdit, EditEventDescriptor editEventDescriptor)
+            throws CommandException {
```
Fix formatting

junhonglow commented on src/main/java/seedu/address/logic/commands/EditContactEventCommand.java:
```diff
@@ -0,0 +1,270 @@
+    public static final String MESSAGE_USAGE = COMMAND_WORD + ": Edits the details of the identified person's calendar "
+            + "by the index number used in the displayed person list. "
+            + "Existing values will be overwritten by the input values.\n"
+            + "Parameters: INDEX (must be a positive integer) "
+            + "[" + PREFIX_EVENT_DESCRIPTION + "DESCRIPTION] "
+            + "[" + PREFIX_EVENT_START_DATE_TIME + "START_DATE] "
+            + "[" + PREFIX_EVENT_END_DATE_TIME + "END_DATE] \n"
```
Missing Event Index Parameter

Fallman2 reviewed (approved):
Check formatting issues

nicrandomlee committed:
Fix bugs

Fix bug where success message incorrectly displayed.
Fix bug where message usage incorrectly displayed.
Updated test case.

nicrandomlee committed:
Fix message usage error

nicrandomlee committed:
Fix Checkstyle Violations

nicrandomlee committed:
Fix bug

Removed irrelevant step in test case

nicrandomlee committed:
Revert "Fix message usage error"

This reverts commit c7489a8f6b70576557cd6c0a2c1a816b29f4cc12.

nicrandomlee committed:
Reverse Commit

nicrandomlee committed:
Fix test case

nicrandomlee committed:
Fix bug in EditContactEventCommand message usage

nicrandomlee committed:
Fix Checkstyle Violations

Removed unused imports

nicrandomlee committed:
Merge branch 'master' into branch-Misc-Calendar-Commands-NL

junhonglow committed:
Fix code formatting issues
